### PR TITLE
Feature toggle

### PR DESCRIPTION
**Pull request by jon-shipley**
Feature toggle: 

* Create new group (19714)
* Remove group (19715)
* Edit Group
* Add pupil
* Edit pupil
* Upload pupils

### CODE REVIEW

File: admin/config/feature-toggles.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+'use strict'
+
+/**
+ * List of feature to toggle on or off
+ * NB features should be written as a positive
+ * @type {{groupCreate: boolean}}
+ */
+
+module.exports = {
```
Comment (GuyHarwood) [inline admin/config/feature-toggles.js]: as discussed - maybe do this as JSON, and we can have files such as....
`feature-toggles.dev.json`
`feature-toggles.uat.json`
etc
Add new env var to config.js so the target file can be overriden at runtime for each environment.
maybe consider employing ramda for merging overrides?

File: admin/app.js [hunk 2]
```diff
@@ -22,11 +23,41 @@ const helmet = require('helmet')
 const config = require('./config')
 const devWhitelist = require('./whitelist-dev')
 const azure = require('./azure')
+const featureToggles = require('feature-toggles')
+const winston = require('winston')
+
+winston.info('ENVIRONMENT_NAME : ' + process.env.ENVIRONMENT_NAME)
+const environmentName = process.env.ENVIRONMENT_NAME
```
Comment (GuyHarwood) [inline admin/app.js]: this value is in the config.js file, please source from there

File: admin/app.js [hunk 3]
```diff
@@ -22,11 +23,41 @@ const helmet = require('helmet')
...
+try {
+  featureTogglesSpecificPath = './config/feature-toggles-' + environmentName
+  featureTogglesDefaultPath = './config/feature-toggles-default'
+  featureTogglesSpecific = environmentName ? require(featureTogglesSpecificPath) : null
+  featureTogglesDefault = require(featureTogglesDefaultPath)
+} catch (error) {
+}
+
+if (featureTogglesSpecific) {
+  winston.info(`Loading environment-specific feature toggles from '${featureTogglesSpecificPath}' : `, featureTogglesSpecific)
```
Comment (GuyHarwood) [inline admin/app.js]: please perform a merge with the defaults here.  OR does the package take care of that for you, by applying them in order?

File: admin/config/feature-toggles-Azure_UAT.json [hunk 4]
```diff
@@ -1,4 +1,7 @@
 {
   "groupCreate": false,
-  "groupRemove": false
+  "groupRemove": false,
```
Comment (GuyHarwood) [inline admin/config/feature-toggles-Azure_UAT.json]: at the risk of sounding pedantic, can we use a dot, rather than a comma in the filenames please?

Comment (jon-shipley) [reply admin/config/feature-toggles-Azure_UAT.json]: Yes, sure.

**Pull request merged**